common/lib/xmodule/xmodule/js/src/video/10_main.js

```diff
@@ -138,6 +140,10 @@
                     initialize(state, element);
                 }
 
+                if (autoAdvanceEnabled) {
+                    mainVideoModules.push(VideoAutoAdvanceControl);
+                }
```

Comment:
There's the style convention that „variables must be defined at the top“ (https://eslint.org/docs/rules/vars-on-top). If I did that I would break this rule, because there are more vars after line 90.
I saw that there's {{state}} using {{mainVideoModules}} in it, but my code is still safe because arrays are passed by reference (http://www.dyn-web.com/javascript/arrays/value-vs-reference.php), so I can assign it first to {{state}} and modify the array later, and {{state}} will see the new one. But that's what I meant when I said I didn't like this approach! It's complex or ugly. I now did another change to follow your request without breaking the „vars-on-top“ rule. Tested and pushed.